aten/src/ATen/native/cuda/RenormKernel.cu

```diff
@@ -0,0 +1,29 @@
+#include <ATen/native/Normalization.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cuda/Loops.cuh>
+
+#include <ATen/Dispatch.h>
+
+namespace at {
+namespace native{
+namespace {
+
+void renorm_scale_factor_impl(TensorIteratorBase& iter, double maxnorm) {
+  AT_DISPATCH_FLOATING_TYPES_AND(kHalf, iter.common_dtype(), "renorm_scale_factor_cpu", [&] {
```

Comment:
iter.common_dtype cannot be kHalf, right? It would be kFLoat since `norm` is float